test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2054,6 +2054,31 @@ public void ensurePeerRecoveryRetentionLeasesRenewedAndSynced(String index) thro
         }, 60, TimeUnit.SECONDS);
     }
 
+    private static Map<String, Set<String>> getClusterStateFeatures() throws IOException {
+        final Request request = new Request("GET", "_cluster/state");
+        request.addParameter("filter_path", "nodes_features");
+
+        final Response response = adminClient().performRequest(request);
+
+        Map<String, Set<String>> clusterFeatures = new HashMap<>();
+
+        var responseData = responseAsMap(response);
+        if (responseData.containsKey("nodes_features")) {
```

Comment:
nit - the `containsKey` is superfluous, you can just call `get` and check for null

Reply:
Yes, better. Should not make a huge difference but it's a better habit, will change that.